packages/router/src/router.ts

```diff
@@ -730,6 +732,27 @@ export class Router {
                                 * current "settled" URL. This way the next navigation will be coming
                                 * from the current URL in the browser.
                                 */
+                               if (ignoreSameUrlNavigation) {
+                                 let reason = '';
+                                 if (NG_DEV_MODE) {
+                                   reason = `Navigation to ${
+                                       t.rawUrl} was ignored because it is the same as the current Router URL.`;
+                                 }
+                                 this.triggerEvent(new NavigationCancel(
+                                     t.id, this.serializeUrl(overallTransitionState.rawUrl), reason,
+                                     NavigationCancellationCode.IgnoredSameUrlNavigation));
+                               } else {
+                                 let reason = '';
+                                 if (NG_DEV_MODE) {
+                                   reason =
+                                       `Navigation was ignored because the UrlHandlingStrategy indicated neither the current URL ${
+                                           this.rawUrlTree} nor target URL ${
+                                           t.rawUrl} should be processed.`;
+                                 }
+                                 this.triggerEvent(new NavigationCancel(
+                                     t.id, this.serializeUrl(overallTransitionState.extractedUrl),
+                                     reason, NavigationCancellationCode.IgnoredSameUrlNavigation));
```

Comment:
We should probably have a different code here?
```suggestion
                                     reason, NavigationCancellationCode.IgnoredByUrlHandlingStrategy));
```